src/Common/src/Interop/OSX/System.Security.Cryptography.Native.Apple/Interop.X509.cs

```diff
@@ -117,26 +125,50 @@ internal static byte[] X509GetRawData(SafeSecCertificateHandle cert)
             bool exportable,
             out SafeSecIdentityHandle identityHandle)
         {
-            SafeSecCertificateHandle certHandle;
-            int osStatus;
-            int ret;
-
-            SafeCreateHandle cfPassphrase = s_nullExportString;
+            SafeCreateHandle cfPassphrase = null;
             bool releasePassword = false;
 
             try
             {
                 if (!importPassword.IsInvalid)
                 {
                     importPassword.DangerousAddRef(ref releasePassword);
-                    IntPtr passwordHandle = importPassword.DangerousGetHandle();
+                    cfPassphrase = CoreFoundation.CFStringCreateFromSpan(importPassword.DangerousGetSpan());
```

Comment:
should this handle be freed after construction or will it be owned by X509ImportCertificate?

Reply:
In the code here it's disposed by the called method, I've now moved the dispose back to this method.